### ISSUE TITLE
Rebase the kernel based on @torvalds/linux

### ISSUE THREAD

**Issue by msva** [253647445/Pocket-Ubuntu-kernel#1]
Would you be so kind to rebase the kernel in this repo basing on [upstream repo](https://github.com/torvalds/linux) instead of current single-mega-commit way, please?

**Comment by Jbonnett** [comment]
The current state leaves it to us to figure out what you changed and figure out how to port it to the kernels we use, which is terrible. Please generate a diff and commit it in logical chunks.

**Comment by eMBee** [comment]
for starters, please at least tell us which original source you used.

did you use one of the methods described here?
https://wiki.ubuntu.com/Kernel/SourceCode

**Comment by Herst** [comment]
Maybe the patch files could be checked in as well.

**Comment by Jbonnett** [comment]
sorry herst but that isnt how git works, git should hold the diffs.  they
should put patches somewhere else, even a different git repo.

**Comment by Li-Weiwei** [comment]
This repo looks nobody to maintain.  
Please visit http://tieba.baidu.com/p/5293185138. This is where official communication is.

**Comment by icebal** [comment]
Requesting that atleast a changelog ia put up here, to better guide development

**Comment by hamishcoleman** [comment]
I did most of what you are looking for - if you pull https://github.com/hamishcoleman/Pocket-Ubuntu-kernel and then diff the upstream branch to the archeology you will see the diffs for their kernel. (eg with `git diff upstream..archeology`)

However, be aware that this kernel is not recommended for use by the linux kernel hackers working on this style of hardware - see https://www.reddit.com/r/GPDPocket/comments/6ug8e4/do_not_use_the_gpd_pockets_ubuntu_kernel_it_is/ for some discussion

**Comment by Herst** [comment]
@LiWeiwei-GitHub They again linked to this repository on Indiegogo in a campaign update.

**Comment by hamishcoleman** [comment]
Yes, It was that update that reminded me to come back to this repo and make sure I had made some kind of comment

**Comment by maxengel** [comment]
I encourage everyone to check out and contribute to this project: https://github.com/stockmind/gpd-pocket-ubuntu-respin
